### DISCUSSION TITLE
Ask for File/Heading Option.

### DISCUSSION THREAD

**Discussion by SabinaKu** [chhoumann/quickadd#376] [Ideas]
Hello,

firstly thanks for this wonderful plugin.
I would like to use a more flexible QuickAdd Capture. What I mean is: I would like to use a Hotkey. Then QuickAdd asks me what note (and maybe under what heading) I would like to capture something and then asks what text I would like to capture. 
I dont know if you  guys can just add the button "Capture to: ask for file" in the settings of a capture. 

That would be epic!

Thanks

**Comment by chhoumann** [comment]
Hi @SabinaKu 

A while back I made a macro called `Capture Anywhere`. This will prompt for a file to capture to, and then use a capture to capture to it.
First, make a macro with this script:
```js
module.exports = async (params) => {
    const allMdFiles = params.app.vault.getMarkdownFiles();
    const selectedFilePath = await params.quickAddApi.suggester(
        allMdFiles.map(f => f.basename),
        allMdFiles.map(f => f.path)
    );

    params.variables.path = selectedFilePath;
}
```

Then create a capture inside that macro. The file path should be `{{VALUE:path}}`. That is what we use the script earlier for.
Now you can otherwise customize the capture however you like.

**Reply by SabinaKu** [reply]
Thank you for posting this. Im not yet comfortable with using/editing macros. I found a workaround/better option using Apple‘s Shortcut though.
Still would love to see the feature in QuickAdd!

**Reply by chhoumann** [reply]
I've added a feature request in #388 :)

**Comment by chhoumann** [comment]
Coming back to this very late - sorry about that. The good news: this is built in now (#388 shipped), no macro needed.

Leave the Capture To field empty (or type `/`) and QuickAdd asks you which note to capture to when you run it, with a picker over the whole vault. Type a name that doesn't exist and it creates the note for you (with _Create file if it doesn't exist_ enabled). A folder path scopes the picker to that folder, and `#tag` or `property:field=value` filter it further.

For the heading part: under _Insert after_ there's a _Choose heading when capturing_ toggle - QuickAdd then shows a dropdown of the target note's headings at run time and inserts under the one you pick.

Docs: https://quickadd.obsidian.guide/docs/Choices/CaptureChoice#how-quickadd-picks-a-target

Closing as resolved.
